src/app/multi-editor/multi-editor.component.ts

```diff
@@ -61,23 +61,47 @@ export class MultiEditorComponent implements OnInit {
   }
 
   onSave() {
-    this.queryService.save(this.previewedActions, this.checkedRecords)
-    .then((res) => {
-      this.successMessage = res['message'];
-      this.totalRecords = -1;
-      this.changeDetectorRef.markForCheck();
-    })
+    let uuids = Object.keys(this.recordSelectionStatus)
+      .filter(key => this.recordSelectionStatus[key] !== this.allSelected);
+    this.queryService.save(this.previewedActions, uuids, this.allSelected)
       .catch((error) => {
         if (error.json().message) {
           this.totalRecords = -1;
           this.errorMessage = error.json().message;
-        }else {
+        } else {
           this.errorMessage = error;
         }
         this.changeDetectorRef.markForCheck();
       });
   }
 
+  selectAll() {
+    this.allSelected = true;
+    Object.keys(this.recordSelectionStatus)
+      .forEach((item) => { this.recordSelectionStatus[item] = true; });
+    this.changeDetectorRef.markForCheck();
+  }
+
+  deselectAll() {
+    this.allSelected = false;
+    Object.keys(this.recordSelectionStatus)
+      .forEach((item) => { this.recordSelectionStatus[item] = false; });
+    this.changeDetectorRef.markForCheck();
+  }
+
+  private setSelectionStatusesForNewPageRecords() {
+    if (this.recordSelectionStatus.hasOwnProperty(this.uuids[0])) {
```

Comment:
` if (!this.recordSelectionStatus.hasOwnProperty(this.uuids[0]))` and drop the `else`